openmsupply/application-manager-web-app
# #442-part-2-re-review-in-review-page

Needs back end: #442f-add-status-to-application-response, PR: https://github.com/openmsupply/application-manager-server/pull/267

This PR fixes and issue described in part of of https://github.com/openmsupply/application-manager-web-app/issues/442

See testing

### Changes

* application response to have status (it's default as draft on creation on back end), and changed to submitted on submission of application

### Testing

To replicate the bug, switch to #443-show-in-review-home-re-submission-of-application.

* create LOQ review for ABC123
* keep the review summary page open after re-submission
* start re-submission for applicant (dont' submit)
* refresh the review summary page (you will see the review responses disapearing)

Do the same in this branch and you should see review summary page not affected by start of applicant's resubmission


### Notes

This problem could have been fixed by row level permissions, but it seems to be better to build some resilience in the system (even if row level permission are incorrect), similar to isVisibleToApplicant being a filter on application response in getAllReasponses (could be done with row level permission, but it's also good to add it to the query)
